src/test_compose.py

```diff
@@ -0,0 +1,375 @@
+import numpy as np
+from numpy.typing import NDArray
+from typing import Any
+from bmipy import Bmi
+from pathlib import Path
+from enum import Enum
+
+
+class CouplingType(Enum):
+  ONE_WAY = 1
+  TWO_WAY = 2
+  
+
+
+## the idea here is that the interface list passed to the compose function would be in a specified order, so interface[0] is a dictionary with units that
+## the user wishes to set during the update, interface[1] could be conversions but I think this may just need to be hard coded by the user outside of the update
+## as in the gipl example the conversion is not done shared variables but rather 2 different ones.
+
+def compose(bmi1 : Bmi, bmi2 : Bmi, coupling_type : CouplingType = CouplingType.ONE_WAY, interface : list[str] = []) -> Bmi:
+
+  if coupling_type == CouplingType.TWO_WAY:
+    interfaceVars1 = intersection(bmi1.get_output_var_names(), bmi2.get_input_var_names())
+    interfaceVars2 = intersection(bmi1.get_input_var_names(), bmi2.get_output_var_names())
+  
+  
+  else:
+    interfaceVars1 = intersection(bmi1.get_output_var_names(), bmi2.get_input_var_names())
+
+  
+
+
+
+  class ComposedBmi(Bmi):
+
+    def setup(self, *args, **kwargs):
+      
+      defaults1 = bmi1.setup(*args, **kwargs)
+      defaults2 = bmi2.setup(*args, **kwargs)
+
+      print(defaults1)
+      
+      defaults = composeConfig(defaults1[1] + "/" + defaults1[0], defaults2[1] + "/" + defaults2[0])
```

Comment:
Needs to be made architecture agnostic as paths with "/" are UNIX specific. I think you want something like:
```
path1 = pathlib.Path(*defaults)
```